Backend : Getting 500 internal server error for follows activity feed filter

**Affected module**
Does it impact the UI, backend or Ingestion Framework?
Backend

**Describe the bug**
A clear and concise description of what the bug is.
When we try to filter the activity feed for the following data backend is throwing a 500 internal server error.
```
{
    "responseMessage": "An exception with message [Error rendering SQL template: 'SELECT json FROM thread_entity WHERE updatedAt < :after AND resolved = :resolved AND entityId in (SELECT toId FROM entity_relationship WHERE ((fromEntity='user' AND fromId= :userId) OR (fromEntity='team' AND fromId IN (<teamIds>))) AND relation= :relation) ORDER BY updatedAt DESC LIMIT :limit' [statement:\"null\", arguments:{positional:{0:a19352d4-4db3-49c6-8f12-32c4e2c5d8be,1:11,2:9223372036854775807,3:false,4:11}, named:{limit:11,after:9223372036854775807,userId:a19352d4-4db3-49c6-8f12-32c4e2c5d8be,resolved:false,relation:11}, finder:[]}]] was thrown while processing request."
}
```

Server Logs

```
ERROR [09:36:36.218] [dw-37 - GET /api/v1/feed?userId=15662f70-1a41-4d35-9a09-35fe71ca2b09&filterType=FOLLOWS] o.o.c.r.f.FeedResource - Got exception: [UnableToCreateStatementException] / message [Error rendering SQL template: 'SELECT json FROM thread_entity WHERE updatedAt < :after AND resolved = :resolved AND entityId in (SELECT toId FROM entity_relationship WHERE ((fromEntity='user' AND fromId= :userId) OR (fromEntity='team' AND fromId IN (<teamIds>))) AND relation= :relation) ORDER BY updatedAt DESC LIMIT :limit' [statement:"null", arguments:{positional:{0:15662f70-1a41-4d35-9a09-35fe71ca2b09,1:11,2:9223372036854775807,3:false,4:11}, named:{limit:11,after:9223372036854775807,userId:15662f70-1a41-4d35-9a09-35fe71ca2b09,resolved:false,relation:11}, finder:[]}]] / related resource location: [org.openmetadata.catalog.resources.feeds.FeedResource.list](FeedResource.java:187) 
org.jdbi.v3.core.statement.UnableToCreateStatementException: Error rendering SQL template: 'SELECT json FROM thread_entity WHERE updatedAt < :after AND resolved = :resolved AND entityId in (SELECT toId FROM entity_relationship WHERE ((fromEntity='user' AND fromId= :userId) OR (fromEntity='team' AND fromId IN (<teamIds>))) AND relation= :relation) ORDER BY updatedAt DESC LIMIT :limit' [statement:"null", arguments:{positional:{0:15662f70-1a41-4d35-9a09-35fe71ca2b09,1:11,2:9223372036854775807,3:false,4:11}, named:{limit:11,after:9223372036854775807,userId:15662f70-1a41-4d35-9a09-35fe71ca2b09,resolved:false,relation:11}, finder:[]}]
	at org.jdbi.v3.core.statement.DefinedAttributeTemplateEngine.lambda$parse$4(DefinedAttributeTemplateEngine.java:92)
	at org.jdbi.v3.core.statement.TemplateEngine$Parsing.render(TemplateEngine.java:67)
	at io.dropwizard.jdbi3.NamePrependingTemplateEngine.render(NamePrependingTemplateEngine.java:26)
	at org.jdbi.v3.core.statement.SqlStatements.lambda$preparedRender$2(SqlStatements.java:277)
	at org.jdbi.v3.core.statement.SqlStatements.preparedRender(SqlStatements.java:278)
	at org.jdbi.v3.core.statement.SqlStatement.parseSql(SqlStatement.java:1808)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1763)
	at org.jdbi.v3.core.result.ResultProducers.lambda$getResultSet$2(ResultProducers.java:64)
	at org.jdbi.v3.core.result.ResultIterable.lambda$of$0(ResultIterable.java:54)
	at org.jdbi.v3.core.result.ResultIterable.stream(ResultIterable.java:228)
	at org.jdbi.v3.core.result.ResultIterable.collect(ResultIterable.java:284)
	at org.jdbi.v3.sqlobject.statement.internal.ResultReturner$CollectedResultReturner.mappedResult(ResultReturner.java:275)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.lambda$configureReturner$0(SqlQueryHandler.java:61)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:187)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.lambda$invoke$0(SqlObjectInitData.java:132)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:44)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:26)
	at org.jdbi.v3.core.LazyHandleSupplier.lambda$invokeInContext$1(LazyHandleSupplier.java:79)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:44)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:26)
	at org.jdbi.v3.core.LazyHandleSupplier.invokeInContext(LazyHandleSupplier.java:78)
	at org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.call(SqlObjectInitData.java:138)
	at org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.invoke(SqlObjectInitData.java:132)
	at org.jdbi.v3.sqlobject.SqlObjectFactory.lambda$attach$2(SqlObjectFactory.java:110)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$invoke$5(OnDemandExtensions.java:98)
	at org.jdbi.v3.core.internal.exceptions.Unchecked.lambda$function$4(Unchecked.java:76)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:98)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:82)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:491)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:478)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:82)
	at com.sun.proxy.$Proxy167.listThreadsByFollowsAfter(Unknown Source)
	at org.openmetadata.catalog.jdbi3.FeedRepository.getThreadsByFollows(FeedRepository.java:556)
	at org.openmetadata.catalog.jdbi3.FeedRepository.list(FeedRepository.java:367)
	at org.openmetadata.catalog.resources.feeds.FeedResource.list(FeedResource.java:187)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:475)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:397)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:366)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:319)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1631)
	at io.dropwizard.servlets.ThreadNameFilter.doFilter(ThreadNameFilter.java:35)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at io.dropwizard.jersey.filter.AllowedMethodsFilter.handle(AllowedMethodsFilter.java:47)
	at io.dropwizard.jersey.filter.AllowedMethodsFilter.doFilter(AllowedMethodsFilter.java:41)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1434)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1349)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at com.codahale.metrics.jetty9.InstrumentedHandler.handle(InstrumentedHandler.java:313)
	at io.dropwizard.jetty.RoutingHandler.handle(RoutingHandler.java:52)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:763)
	at org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:54)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:179)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:400)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:645)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:392)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.jdbi.v3.core.statement.UnableToCreateStatementException: Undefined attribute for token '<teamIds>' [statement:"null", arguments:{positional:{0:15662f70-1a41-4d35-9a09-35fe71ca2b09,1:11,2:9223372036854775807,3:false,4:11}, named:{limit:11,after:9223372036854775807,userId:15662f70-1a41-4d35-9a09-35fe71ca2b09,resolved:false,relation:11}, finder:[]}]
	at org.jdbi.v3.core.statement.DefinedAttributeTemplateEngine.lambda$parse$2(DefinedAttributeTemplateEngine.java:72)
	at org.jdbi.v3.core.statement.DefinedAttributeTemplateEngine.lambda$parse$3(DefinedAttributeTemplateEngine.java:89)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.jdbi.v3.core.statement.DefinedAttributeTemplateEngine.lambda$parse$4(DefinedAttributeTemplateEngine.java:89)
	... 101 common frames omitted

```



**To Reproduce**
- Go to the My data page 
- Try to filter the activity feed for the following data.

**Expected behavior**
API should return activity feed for follows filter (if any) or an empty list.

--- a/catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java
+++ b/catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java
@@ -654,6 +654,7 @@ public interface CollectionDAO {
             + "LIMIT :limit")
     List<String> listThreadsByFollowsBefore(
         @Bind("userId") String userId,
+        @BindList("teamIds") List<String> teamIds,
         @Bind("limit") int limit,
         @Bind("before") long before,
         @Bind("resolved") boolean resolved,
@@ -668,6 +669,7 @@ public interface CollectionDAO {
             + "LIMIT :limit")
     List<String> listThreadsByFollowsAfter(
         @Bind("userId") String userId,
+        @BindList("teamIds") List<String> teamIds,
         @Bind("limit") int limit,
         @Bind("after") long after,
         @Bind("resolved") boolean resolved,
@@ -679,7 +681,10 @@ public interface CollectionDAO {
             + "((fromEntity='user' AND fromId= :userId) OR "
             + "(fromEntity='team' AND fromId IN (<teamIds>))) AND relation= :relation)")
     int listCountThreadsByFollows(
-        @Bind("userId") String userId, @Bind("resolved") boolean resolved, @Bind("relation") int relation);
+        @Bind("userId") String userId,
+        @BindList("teamIds") List<String> teamIds,
+        @Bind("resolved") boolean resolved,
+        @Bind("relation") int relation);
 
     @SqlQuery(
         "SELECT json FROM thread_entity WHERE updatedAt > :before AND resolved = :resolved AND id in ("

--- a/catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java
+++ b/catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java
@@ -550,13 +550,19 @@ public class FeedRepository {
   private FilteredThreads getThreadsByFollows(
       String userId, int limit, long time, boolean isResolved, PaginationType paginationType) throws IOException {
     List<String> jsons;
+    List<String> teamIds = getTeamIds(userId);
     if (paginationType == PaginationType.BEFORE) {
-      jsons = dao.feedDAO().listThreadsByFollowsBefore(userId, limit, time, isResolved, Relationship.FOLLOWS.ordinal());
+      jsons =
+          dao.feedDAO()
+              .listThreadsByFollowsBefore(userId, teamIds, limit, time, isResolved, Relationship.FOLLOWS.ordinal());
     } else {
-      jsons = dao.feedDAO().listThreadsByFollowsAfter(userId, limit, time, isResolved, Relationship.FOLLOWS.ordinal());
+      jsons =
+          dao.feedDAO()
+              .listThreadsByFollowsAfter(userId, teamIds, limit, time, isResolved, Relationship.FOLLOWS.ordinal());
     }
     List<Thread> threads = JsonUtils.readObjects(jsons, Thread.class);
-    int totalCount = dao.feedDAO().listCountThreadsByFollows(userId, isResolved, Relationship.FOLLOWS.ordinal());
+    int totalCount =
+        dao.feedDAO().listCountThreadsByFollows(userId, teamIds, isResolved, Relationship.FOLLOWS.ordinal());
     return new FilteredThreads(threads, totalCount);
   }
 

--- a/catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/feeds/FeedResourceTest.java
+++ b/catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/feeds/FeedResourceTest.java
@@ -14,6 +14,7 @@
 package org.openmetadata.catalog.resources.feeds;
 
 import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.CREATED;
 import static javax.ws.rs.core.Response.Status.FORBIDDEN;
 import static javax.ws.rs.core.Response.Status.NOT_FOUND;
 import static org.junit.jupiter.api.Assertions.assertEquals;
@@ -519,6 +520,31 @@ public class FeedResourceTest extends CatalogApplicationTest {
   }
 
   @Test
+  void list_threadsWithFollowsFilter() throws HttpResponseException {
+    // Create threads
+    createAndCheck(
+        create().withMessage("Message 1").withAbout(String.format("<#E/table/%s>", TABLE2.getFullyQualifiedName())),
+        ADMIN_AUTH_HEADERS);
+
+    createAndCheck(
+        create().withMessage("Message 2").withAbout(String.format("<#E/table/%s>", TABLE2.getFullyQualifiedName())),
+        ADMIN_AUTH_HEADERS);
+
+    // Make the USER follow TABLE2
+    followTable(TABLE2.getId(), USER.getId(), AUTH_HEADERS);
+
+    ThreadList threads = listThreadsWithFilter(USER.getId().toString(), FilterType.FOLLOWS.toString(), AUTH_HEADERS);
+    assertEquals(2, threads.getPaging().getTotal());
+    assertEquals(2, threads.getData().size());
+    assertEquals("Message 2", threads.getData().get(0).getMessage());
+
+    // Filter by follows for another user should return 0 threads
+    threads = listThreadsWithFilter(USER2.getId().toString(), FilterType.FOLLOWS.toString(), AUTH_HEADERS);
+    assertEquals(0, threads.getPaging().getTotal());
+    assertEquals(0, threads.getData().size());
+  }
+
+  @Test
   void list_threadsWithInvalidFilter() {
     assertResponse(
         () -> listThreadsWithFilter(USER.getId().toString(), "Invalid", AUTH_HEADERS),
@@ -677,6 +703,12 @@ public class FeedResourceTest extends CatalogApplicationTest {
     target = before != null ? target.queryParam("before", before) : target;
     target = after != null ? target.queryParam("after", after) : target;
     return TestUtils.get(target, ThreadList.class, authHeaders);
+  }
+
+  public static void followTable(UUID tableId, UUID userId, Map<String, String> authHeaders)
+      throws HttpResponseException {
+    WebTarget target = getResource("tables/" + tableId + "/followers");
+    TestUtils.put(target, userId.toString(), CREATED, authHeaders);
   }
 
   public static ThreadList listThreadsWithFilter(String userId, String filterType, Map<String, String> authHeaders)

--- a/catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/feeds/FeedResourceTest.java
+++ b/catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/feeds/FeedResourceTest.java
@@ -521,21 +521,21 @@ public class FeedResourceTest extends CatalogApplicationTest {
 
   @Test
   void list_threadsWithFollowsFilter() throws HttpResponseException {
-    // Create threads
-    createAndCheck(
-        create().withMessage("Message 1").withAbout(String.format("<#E/table/%s>", TABLE2.getFullyQualifiedName())),
-        ADMIN_AUTH_HEADERS);
+    // Get the initial thread count of TABLE2
+    String entityLink = String.format("<#E/table/%s>", TABLE2.getFullyQualifiedName());
+    int initialThreadCount = listThreads(entityLink, null, AUTH_HEADERS).getPaging().getTotal();
 
-    createAndCheck(
-        create().withMessage("Message 2").withAbout(String.format("<#E/table/%s>", TABLE2.getFullyQualifiedName())),
-        ADMIN_AUTH_HEADERS);
+    // Create threads
+    createAndCheck(create().withMessage("Message 1").withAbout(entityLink), ADMIN_AUTH_HEADERS);
+
+    createAndCheck(create().withMessage("Message 2").withAbout(entityLink), ADMIN_AUTH_HEADERS);
 
     // Make the USER follow TABLE2
     followTable(TABLE2.getId(), USER.getId(), AUTH_HEADERS);
 
     ThreadList threads = listThreadsWithFilter(USER.getId().toString(), FilterType.FOLLOWS.toString(), AUTH_HEADERS);
-    assertEquals(2, threads.getPaging().getTotal());
-    assertEquals(2, threads.getData().size());
+    assertEquals(initialThreadCount + 2, threads.getPaging().getTotal());
+    assertEquals(initialThreadCount + 2, threads.getData().size());
     assertEquals("Message 2", threads.getData().get(0).getMessage());
 
     // Filter by follows for another user should return 0 threads

--- a/openmetadata-ui/src/main/resources/ui/src/jsons/en.ts
+++ b/openmetadata-ui/src/main/resources/ui/src/jsons/en.ts
@@ -54,6 +54,7 @@ const jsonData = {
     'fetch-data-error': 'Error while fetching data!',
     'fetch-database-details-error': 'Error while fetching database details!',
     'fetch-database-tables-error': 'Error while fetching database tables!',
+    'fetch-activity-feed-error': 'Error while fetching activity feeds!',
     'fetch-entity-feed-error': 'Error while fetching entity feeds!',
     'fetch-entity-feed-count-error': 'Error while fetching entity feed count!',
     'fetch-entity-count-error': 'Error while fetching entity count!',

--- a/openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx
+++ b/openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx
@@ -233,11 +233,13 @@ const MyDataPage = () => {
         const { data } = res.data;
         setEntityThread(data);
       })
-      .catch(() => {
-        showToast({
-          variant: 'error',
-          body: 'Error while fetching the Activity Feed',
-        });
+      .catch((err: AxiosError) => {
+        handleShowErrorToast(
+          getErrorText(
+            err,
+            jsonData['api-error-messages']['fetch-activity-feed-error']
+          )
+        );
       })
       .finally(() => {
         setIsFeedLoading(false);

--- a/openmetadata-ui/src/main/resources/ui/src/utils/StringsUtils.ts
+++ b/openmetadata-ui/src/main/resources/ui/src/utils/StringsUtils.ts
@@ -106,7 +106,17 @@ export const getErrorText = (
   value: AxiosError | string,
   fallbackText: string
 ): string => {
-  return (
-    (isString(value) ? value : value.response?.data?.message) || fallbackText
-  );
+  let errorText;
+  if (isString(value)) {
+    return value;
+  } else {
+    errorText = value.response?.data?.message;
+    if (!errorText) {
+      // if error text is undefined or null or empty, try responseMessage in data
+      errorText = value.response?.data?.responseMessage;
+    }
+  }
+
+  // if error text is still empty, return the fallback text
+  return errorText || fallbackText;
 };

